integration/embed_test.go

```diff
@@ -299,3 +300,136 @@ func embedTestHelper(ctx context.Context, client *api.Client, t *testing.T, req
 
 	return client.Embed(ctx, &req)
 }
+
+func TestEmbedTruncationRefactor(t *testing.T) {
```

Comment:
Assuming that "refactor" in `TestEmbedTruncationRefactor` refers to this change, I think it would be better to leave it out of the name - i.e. just `TestEmbedTruncation`. In the future, you don't want to have to think of the code base as a series of changes but rather something that makes sense logically at the current point in time.